### PR TITLE
Audit Admin Login Flow

### PR DESCRIPTION
**Pull request by benadida**
Implement the Auth0 flow for audit admins.
We may have a separate flow for jurisdiction admins, we'll see.

As per product tickets #250 and #260.

### CODE REVIEW

File: config/__init__.py [hunk 1]
```diff
@@ -55,3 +57,42 @@ def read_session_secret() -> str:
...
+def read_jurisdictionadmin_auth0_creds() -> Tuple[str, str, str]:
+    return (os.environ.get('ARLO_JURISDICTIONADMIN_AUTH0_BASE_URL',
+                           ''), os.environ.get('ARLO_JURISDICTIONADMIN_AUTH0_CLIENT_ID', ''),
+            os.environ.get('ARLO_JURISDICTIONADMIN_AUTH0_CLIENT_SECRET', ''))
+
+
+JURISDICTIONADMIN_AUTH0_BASE_URL, JURISDICTIONADMIN_AUTH0_CLIENT_ID, JURISDICTIONADMIN_AUTH0_CLIENT_SECRET = read_jurisdictionadmin_auth0_creds(
+)
+
+print("ID", JURISDICTIONADMIN_AUTH0_CLIENT_ID)
```
Comment (eventualbuddha) [inline config/__init__.py]: Remove debugging.

File: arlo_server/__init__.py [hunk 2]
```diff
@@ -896,6 +902,116 @@ def incr():
...
+            set_loggedin_user('auditadmin', userinfo['email'])
+
+    return redirect('/')
+
+
+@app.route('/auth/jurisdictionadmin/start')
+def jurisdictionadmin_login():
+    print("YO")
+    print(auth0_ja.client_id)
+    print(auth0_aa.client_id)
```
Comment (eventualbuddha) [inline arlo_server/__init__.py]: Remove debugging.

File: arlo_server/__init__.py [hunk 3]
```diff
@@ -896,6 +902,116 @@ def incr():
...
+@app.route(AUDITADMIN_OAUTH_CALLBACK_URL)
+def auditadmin_login_callback():
+    auth0_aa.authorize_access_token()
+    resp = auth0_aa.get('userinfo')
+    userinfo = resp.json()
+
+    if userinfo and userinfo['email']:
+        user = User.query.filter_by(email=userinfo['email']).first()
+        if user and len(user.audit_administrations) > 0:
+            set_loggedin_user('auditadmin', userinfo['email'])
```
Comment (eventualbuddha) [inline arlo_server/__init__.py]: Any good reason not to have `auditadmin` and `jusisdictionadmin` be enum values?

File: docs/auth0.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+- setting up auth0 passwordless requires either creating users via the
+  Management API, or letting anyone sign in and filtering on our
+  end. We'll start with the latter, we may do the former at some
+  point.
+
+- right now we're using "Universal Login", where Auth0 controls the
+  login page. It's not clear that's the right way forward for Arlo, as
+  customization is limited and we can't unify the login flows or
+  provide error messages at the idea time. We may want the embedded
```
Comment (eventualbuddha) [inline docs/auth0.md]: `at the idea time`?

File: arlo_server/__init__.py (hunk 2 above)
Comment (benadida) [reply arlo_server/__init__.py]: lol, you don't like `print("YO")` ?

**Pull request merged**